evm/vm/base.py

```diff
@@ -188,6 +272,8 @@ def get_block_class(self):
 
     def get_block_by_hash(self, block_hash):
         block_header = self.evm.get_block_header_by_hash(block_hash)
+        if block_header is None:
+            return None
```

Comment:
My *aversion* to this mechanism is it creates the following pattern

```python
block = evm.get_block_by_hash(....)
if block is None:
    # handle it.
...
```

The thing I dislike about this pattern is that it requires knowledge that the return value *may* be `None` and *if* the user doesn't know this, then the error may arise in a very different part of the application.

If instead it raises an exception, then the error is guaranteed to be raised at the point that `get_block_by_hash` was called, and you can still have quite clean handling with `try/except`.

Thoughts?